File: emulinker/src/main/java/org/emulinker/kaillera/model/event/GameDesynchEvent.java
package org.emulinker.kaillera.model.event;

import org.emulinker.kaillera.model.KailleraGame;

public class GameDesynchEvent implements GameEvent {
  private KailleraGame game;
  private String message;

  public GameDesynchEvent(KailleraGame game, String message) {
    this.game = game;
    this.message = message;
  }

  @Override
  public String toString() {
    return "GameDesynchEvent";
  }

  @Override
  public KailleraGame getGame() {
    return game;
  }

  public String getMessage() {
    return message;
  }
}

File: emulinker/src/main/java/org/emulinker/kaillera/model/event/GameDesynchEvent.kt
package org.emulinker.kaillera.model.event

import org.emulinker.kaillera.model.KailleraGame

data class GameDesynchEvent(override val game: KailleraGame, val message: String) : GameEvent {
  override fun toString(): String {
    return "GameDesynchEvent"
  }
}
